trieste/models/gpflux/builders.py

```diff
@@ -154,3 +155,92 @@ def _get_data_stats(data: Dataset) -> tuple[TensorType, TensorType, int]:
+
+    :param data: Dataset from the initial design, used to estimate the variance of observations
+        and to provide query points which are used to determine inducing point locations with
+        k-means.
+    :param search_space: Search space for performing Bayesian optimization. Used for initialization
+        of inducing locations if ``num_inducing_points`` is larger than the amount of data.
+    :param num_layers: Number of layers in deep GP. By default set to ``NUM_LAYERS``.
+    :param num_inducing_points: Number of inducing points to use in each layer. If left unspecified
+        (default), this number is set to either ``NUM_INDUCING_POINTS_PER_DIM``*dimensionality of
+        the search space or value given by ``MAX_NUM_INDUCING_POINTS``, whichever is smaller.
+    :param inner_layer_sqrt_factor: A multiplicative factor used to rescale hidden layers, see
+        :class:`~gpflux.architectures.Config` for details. By default set to
+        ``INNER_LAYER_SQRT_FACTOR``.
+    :param likelihood_variance: Initial noise variance in the likelihood function, see
+        :class:`~gpflux.architectures.Config` for details. By default set to
+        ``LIKELIHOOD_VARIANCE``.
+    :param trainable_likelihood: Trainable likelihood variance.
+    :return: A :class:`~gpflux.models.DeepGP` model with sensible default settings.
+    :raise: If non-positive ``num_layers``, ``inner_layer_sqrt_factor``, ``likelihood_variance``
+        or ``num_inducing_points`` is provided.
+    """
+    tf.debugging.assert_positive(num_layers)
+    tf.debugging.assert_positive(inner_layer_sqrt_factor)
+    tf.debugging.assert_positive(likelihood_variance)
+
+    # Input data to ``build_constant_input_dim_deep_gp`` must be np.ndarray for k-means algorithm
+    query_points = data.query_points.numpy()
+
+    empirical_mean, empirical_variance, num_data_points = _get_data_stats(data)
+
+    if num_inducing_points is None:
+        num_inducing_points = min(
+            MAX_NUM_INDUCING_POINTS, NUM_INDUCING_POINTS_PER_DIM * int(search_space.dimension)
+        )
+    else:
+        tf.debugging.assert_positive(num_inducing_points)
+
+    # Pad query_points with additional random values to provide enough inducing points
+    if num_inducing_points > len(query_points):
+        if isinstance(search_space, Box):
+            additional_points = search_space.sample_sobol(
+                num_inducing_points - len(query_points)
+            ).numpy()
+        else:
+            additional_points = search_space.sample(num_inducing_points - len(query_points)).numpy()
+        query_points = np.concatenate([query_points, additional_points], 0)
+
+    config = Config(
+        num_inducing_points,
+        inner_layer_sqrt_factor,
+        likelihood_variance,
+        whiten=True,  # whiten = False not supported yet in GPflux for this model
+    )
+
+    model = build_constant_input_dim_flexible_deep_gp(query_points, num_layers, config, True, dim_output)
```

Comment:
@hstojic this is the only difference..if you can think of a more efficient way of doing this, that would be grand

Reply:
you said you need this only for the test, right? you can then move the whole thing there?

Reply:
otherwise there are indeed more efficient ways to do this, e.g. having a private function `_build_deep_gp` which takes a function as an input, either  `build_constant_input_dim_flexible_deep_gp ` or `build_constant_input_dim_deep_gp` and then you have two public functions `build_vanilla_flexible_deep_gp` and `build_vanilla_deep_gp` that call the private one with these two different functions